src/main/java/sg/ncl/DataController.java

```diff
@@ -322,6 +322,31 @@ public void getResource(@PathVariable String datasetId,
         }
     }
 
+    @RequestMapping(value = "{datasetId}/resources/{resourceId}/delete", method = RequestMethod.GET)
+    public String removeResource(@PathVariable String datasetId, @PathVariable String resourceId, RedirectAttributes redirectAttributes) throws WebServiceRuntimeException {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        restTemplate.setErrorHandler(new MyResponseErrorHandler());
+        ResponseEntity response = restTemplate.exchange(properties.getResource(datasetId, resourceId), HttpMethod.DELETE, request, String.class);
+        String dataResponseBody = response.getBody().toString();
+
+        try {
+            if (RestUtil.isError(response.getStatusCode())) {
+                MyErrorResource error = objectMapper.readValue(dataResponseBody, MyErrorResource.class);
+                ExceptionState exceptionState = ExceptionState.parseExceptionState(error.getError());
+
+                if (exceptionState == FORBIDDEN_EXCEPTION) {
+                    log.error("Removing of data resource forbidden.");
+                    redirectAttributes.addFlashAttribute(MESSAGE_ATTRIBUTE, error.getMessage());
+                }
```

Comment:
1. During my local tests, I cannot delete a file and there is no error message shown on the UI. 
2. You only catch forbidden exception, which is actually unlikely to happen. 
3. You need to catch more exceptions such as data/resource not found exception, and default exception for all the other types of exceptions.
4. Not found could means that the data info is not available in the database, or the actual file is not found on the file system. Do we need to distinguish them?
5. What is more important is to show users some useful info when things went wrong.